### DISCUSSION TITLE
Csla.Blazor.Client.Authentication.CslaUserService no register service

### DISCUSSION THREAD

**Discussion by AsadCheema** [MarimerLLC/csla#2439] [Questions]
Hi All,

I am trying to create another Blazor App, and getting code sippets from ProjectTracker solution. 
Creating as LogIn.razor component, with this inject statement,
      @inject Csla.Blazor.Client.Authentication.CslaUserService userService
when trying to open Login page, following error occurs i.e.
There is no registered service of type 'Csla.Blazor.Client.Authentication.CslaUserService'
I had checked these lines present in startup.cs, 
       services.AddCsla().WithBlazorServerSupport();

using Csla 5.4.1 and .Net 3.1

Regards.
A. Cheema

**Comment by rockfordlhotka** [comment]
The `WithBlazorServerSupport` method doesn't automatically add any user service. It only adds the services necessary for the `HasPermission` attribute.

You must specify the user service yourself at app startup. Such as:

https://github.com/MarimerLLC/csla/blob/613b8795a85aa0ce32ae96b6e975c74d506df0f6/Samples/ProjectTracker/ProjectTracker.Ui.Blazor.Client/Program.cs#L26

However, and this is important, that `UserService` type is for _Blazor WebAssembly_, not server-side Blazor.


You should also be aware that there's a bug in CSLA when dealing with server-side state. It will be fixed in CSLA 6, because the fix requires _massive_ changes to CSLA to accommodate how server-side Blazor manages user identity. That's being tracked in https://github.com/MarimerLLC/csla/issues/1068.
